src/utils/Scale.js

```diff
@@ -117,28 +117,12 @@ export function scaleEqual(scaleA, scaleB) {
         _.isEqual(scaleA.range(), scaleB.range());
 }
 
-export function indexOfClosestLeftNumberInList(number, sortedList) {
-  if (sortedList.length <= 1) {
-    return 0;
-  }
-
-  const isDescending = sortedList[0] > sortedList[1];
-  // _.sortedIndex works on ascending lists only
-  // so reverse if working with descending list
-  const listCopy = isDescending ? sortedList.slice().reverse() : sortedList;
-
-  // Get lower bound of where number falls
-  let indexForNumber = _.sortedIndex(listCopy, number);
-  if (isDescending && indexForNumber < sortedList.length) {
-    indexForNumber += 1;
-  } else if (!isDescending && indexForNumber > 0) {
-    indexForNumber -= 1;
-  }
-
-  return isDescending
-    ? // If descending, remap indexForNumber to work with descending list
-      Math.abs(indexForNumber - sortedList.length)
-    : indexForNumber;
+export function indexOfClosestNumberInList(number, list) {
+  return list.reduce((closestI, current, i) => {
+    return Math.abs(current - number) < Math.abs(list[closestI] - number)
+      ? i
+      : closestI;
+  }, 0);
```

Comment:
Hi! Thanks for keeping the Reactochart dream alive!

Just FYI - the original reason i wrote it <-- that way was because we were assuming the input data would be sorted (by domain, either ascending or descending). With this assumption, you can use `_.sortedIndex` to do a binary search in `O(log n)` time. With ^the new method here, it works for all data regardless of sort order, but always searches every data point, so it's `O(n)` which can get slow for large datasets.

IIRC, the plan was to eventually allow a `sorted` flag (on the `XYPlot` i guess?) that would tell Reactochart whether the data was sorted or not - it would default false, but if `true` would use tricks like this to get better performance for sorted data. Might be a nice enhancement for the future.

Hope all is well with y'all :)